js/pages/questionnaire.js

```diff
@@ -54,13 +68,20 @@ export const questionnaire = async (moduleId) => {
         if(moduleId === 'ModuleSsn') {
             socialSecurityTemplate(data);
         } else {
-            const responseModules = await fetchDataWithRetry(() => getMySurveys([...new Set([
-                fieldMapping.Module1.conceptId,
-                fieldMapping.Module1_OLD.conceptId,
-                fieldMapping.Biospecimen.conceptId,
-                fieldMapping.ClinicalBiospecimen.conceptId,
-                fieldMapping[moduleId].conceptId
-            ])]));
+            const [responseModules, _ ] = await Promise.all([
+                fetchDataWithRetry(() => getMySurveys([...new Set([
+                    fieldMapping.Module1.conceptId,
+                    fieldMapping.Module1_OLD.conceptId,
+                    fieldMapping.Biospecimen.conceptId,
+                    fieldMapping.ClinicalBiospecimen.conceptId,
+                    fieldMapping[moduleId].conceptId
```

Comment:
If there's no duplicate values, creating a  Set instance is not needed.

Reply:
This was Tony's code -- I looked into the set as well. My interpretation: I *think* the `moduleId` variable (determined by the module the participant clicks) can be a match with the first four variables or one of the modules that gets conditionally activated. In case of a match, it avoids duplicates.